src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/visits/PrisonVisitsService.kt

```diff
@@ -1,61 +1,107 @@
+  private val mappingService: MappingService,
   private val telemetryClient: TelemetryClient
 ) {
 
+  private companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
   fun createVisit(visitBookedEvent: VisitBookedEvent) {
     visitsApiService.getVisit(visitBookedEvent.visitId).run {
-      nomisApiService.createVisit(
-        CreateVisitDto(
-          offenderNo = this.prisonerId,
-          prisonId = this.prisonId,
-          startDateTime = LocalDateTime.of(this.visitDate, this.startTime),
-          endTime = this.endTime,
-          visitorPersonIds = this.visitors.map { it -> it.nomisPersonId },
-          issueDate = visitBookedEvent.bookingDate,
-          visitType = "SCON", // TODO mapping
-          // visitRoomId = this.visitRoom,
-          vsipVisitId = this.visitId
-        )
-      )
-      val telemetryProperties = mapOf(
-        "offenderNo" to this.prisonerId,
-        "prisonId" to this.prisonId,
-        "visitId" to this.visitId,
-        "startDateTime" to LocalDateTime.of(this.visitDate, this.startTime).format(DateTimeFormatter.ISO_DATE_TIME),
-        "endTime" to this.endTime.format(DateTimeFormatter.ISO_TIME),
+
+      val telemetryMap = mutableMapOf(
+        "offenderNo" to prisonerId,
+        "prisonId" to prisonId,
+        "visitId" to visitId,
+        "startDateTime" to LocalDateTime.of(visitDate, startTime).format(DateTimeFormatter.ISO_DATE_TIME),
+        "endTime" to endTime.format(DateTimeFormatter.ISO_TIME),
       )
-      telemetryClient.trackEvent("visit-booked-event", telemetryProperties, null)
+
+      if (mappingService.getMappingGivenVsipId(visitBookedEvent.visitId) != null) {
+        telemetryClient.trackEvent("visit-booked-get-map-failed", telemetryMap)
+        log.warn("Mapping already exists for VSIP id $visitId")
+        return
+      }
+
+      val nomisId = try {
+        nomisApiService.createVisit(
+          CreateVisitDto(
+            offenderNo = this.prisonerId,
+            prisonId = this.prisonId,
+            startDateTime = LocalDateTime.of(this.visitDate, this.startTime),
+            endTime = this.endTime,
+            visitorPersonIds = this.visitors.map { it -> it.nomisPersonId },
+            issueDate = visitBookedEvent.bookingDate,
+            visitType = "SCON", // TODO mapping
+          )
+        )
+      } catch (e: Exception) {
+        telemetryClient.trackEvent("visit-booked-create-failed", telemetryMap)
+        log.warn("Unexpected exception", e)
+        return
+      }
+
+      val mapWithNomisId = telemetryMap.plus(Pair("nomisVisitId", nomisId!!))
+
+      try {
+        mappingService.createMapping(
+          MappingDto(nomisId = nomisId, vsipId = visitBookedEvent.visitId, mappingType = "ONLINE")
+        )
+      } catch (e: Exception) {
+        telemetryClient.trackEvent("visit-booked-create-map-failed", mapWithNomisId)
+        log.warn("Unexpected exception", e)
+        return
```

Comment:
yep, makes sense not to go on DLQ (very minor I suspect that should be logged as error)